### DISCUSSION TITLE
i want to add a button in the Admin-UI, how should I proceed?

### DISCUSSION THREAD

**Discussion by zhangwelljan** [ServiceStack/Discuss#58] [Q&A] [answered]
like this
![Snipaste_2023-04-12_13-12-11](https://user-images.githubusercontent.com/49714993/231356107-6bbc2bee-c7cb-49f9-8121-ca430d502b89.png)

**Comment by mythz** [comment, marked as answer]
That doesn't look like it's from the Admin UI, instead from API Explorer or Locode with links to the Admin UI.

To place it exactly where you want it you could redefine the [Welcome.mjs](https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/tests/NorthwindAuto/ui/components/Welcome.mjs) by placing a local modified copy as documented in [Custom Components](https://docs.servicestack.net/locode/custom-components) which is done in [Chinook](https://github.com/NetCoreApps/Chinook/blob/main/Chinook/wwwroot/modules/locode/components/Welcome.mjs).

To override the Welcome component of both API Explorer [Welcome.mjs](https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/tests/NorthwindAuto/ui/components/Welcome.mjs) and Locode [Welcome.mjs](https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/tests/NorthwindAuto/locode/components/Welcome.mjs) you'll need to add local modified copies in your project at:

```
/wwwroot/modules/
    /locode/component/Welcome.mjs
    /ui/component/Welcome.mjs
```

Otherwise the component does support adding "Admin Links" at the bottom which is done by all [Admin UI Features](https://docs.servicestack.net/admin-ui#managing-users-ui) to add links to different Admin Features, e.g [AdminDatabaseFeature.cs](https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/src/ServiceStack.Server/AdminDatabaseFeature.cs#L80) and [AdminRedisFeature.cs](https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/src/ServiceStack/AdminRedisFeature.cs#L63):

```csharp
appHost.ConfigurePlugin<UiFeature>(feature => {
    feature.AddAdminLink(AdminUiFeature.Database, new LinkInfo {
        Id = "database",
        Label = "Database",
        Icon = Svg.ImageSvg(Svg.Create(Svg.Body.Database)),
        Show = $"role:{AdminRole}",
    });
});
```

**Reply by zhangwelljan** [reply]
Thank you! Your answer has given me direction, and I've been looking for a solution to make modifications.
